lib/generators/environment/environment.generator.ts

```diff
@@ -0,0 +1,348 @@
+        readonly collections: readonly Readonly<CollectionModels.Collection>[];
+        readonly roles: readonly Readonly<RoleModels.Role>[];
+        readonly snippets: readonly Readonly<ContentTypeSnippetModels.ContentTypeSnippet>[];
+        readonly webhooks: readonly Readonly<WebhookModels.Webhook>[];
+    };
+}
+
+export function environmentGenerator(config: ProjectGeneratorConfig) {
+    const generateEnvironmentModels = (): readonly GeneratedFile[] => {
+        return [
+            {
+                text: `export const languages = {
+                    ${getProjectLanguages(config.environmentData.languages)}
+                } as const;`,
+                filename: 'languages.ts'
+            },
+            {
+                text: `export const collections = {
+                    ${getCollections(config.environmentData.collections)}
+                } as const;`,
+                filename: 'collections.ts'
+            },
+            {
+                text: `export const contentTypes = {
+                    ${getProjectContentTypes(config.environmentData.types)}
+                } as const;`,
+                filename: 'contentTypes.ts'
+            },
+            {
+                text: `export const contentTypeSnippets = {
+                    ${getProjectContentTypeSnippets(config.environmentData.snippets)}
+                } as const;`,
+                filename: 'contentTypeSnippets.ts'
+            },
+            {
+                text: `export const taxonomies = {
+                    ${getProjectTaxonomies(config.environmentData.taxonomies)}
+                } as const;`,
+                filename: 'taxonomies.ts'
+            },
+            {
+                text: `export const workflows = {
+                    ${getProjectWorkflows(config.environmentData.workflows)}
+                } as const;`,
+                filename: 'workflows.ts'
+            },
+            {
+                text: `export const roles = {
+                    ${getRoles(config.environmentData.roles)}
+                } as const;`,
+                filename: 'roles.ts'
+            },
+            {
+                text: `export const assetFolders = ${getAssetFolders(config.environmentData.assetFolders)} as const;`,
+                filename: 'assetFolders.ts'
+            },
+            {
+                text: `export const webhooks = {
+                    ${getWebhooks(config.environmentData.webhooks)}
+                } as const;`,
+                filename: 'webhooks.ts'
+            }
+        ];
+    };
+
+    const getProjectLanguages = (languages: readonly Readonly<LanguageModels.LanguageModel>[]): string => {
+        return languages.reduce((code, language, index) => {
+            const isLast = index === languages.length - 1;
+
+            return `${code}\n
+                ${wrapComment(`
+                * ${language.name}
+                `)}
+                ${language.codename}: {
```

Comment:
Hello!

I have been fiddling with 8.0.0 in tandem with Delivery SDK 15.0.0 when I came across an issue with language type generation.
Languages allow just about any character in their codenames. I'm assuming this is historical carryover. In my use case, I am using codenames such as `en-US` or `de-DE`. So, when `languages.ts` is generated, I end up with invalid syntax.

Example:
```ts
de-DE: { // <= this needs to be 'de-DE'
  name: 'Deutsch DE',
  codename: 'de-DE',
  id: 'e8929d35-290d-4be8-85a8-f5fe0709dcf9',
  isActive: true,
  isDefault: false,
  fallbackLanguageId: '00000000-0000-0000-0000-000000000000',
  externalId: undefined,
}
```

Reply:
Thank you, I just pushed a fix ;)